src/executionFiles/bridges/hop.ts

```diff
@@ -3,117 +3,101 @@ import {
+  [ChainId.DAI]: 'xdai',
+  [ChainId.OPT]: 'optimism',
+  [ChainId.ARB]: 'arbitrum',
 
   // Testnet; Hop SDK changes the underlying id of these chains according to the instance network
   //network 'goerli'
-  [getChainByKey(ChainKey.GOR).id]: Chain.Ethereum,
-  [getChainByKey(ChainKey.MUM).id]: Chain.Polygon,
+  [ChainId.GOR]: 'ethereum',
+  [ChainId.MUM]: 'polygon',
 }
 
-const supportedTestnetChains: number[] = [ChainId.GOR, ChainId.MUM]
+const getSubgraphUrl = (chainId: ChainId): string => {
+  const chain = hopChains[chainId]
 
-// get these from https://github.com/hop-protocol/hop/blob/develop/packages/sdk/src/models/Token.ts
-const hopTokens: { [k: string]: string } = {
-  USDC: Token.USDC,
-  USDT: Token.USDT,
-  MATIC: Token.MATIC,
-  DAI: Token.DAI,
-  ETH: Token.ETH,
-}
-const isInitialized = () => {
-  if (hop === undefined)
-    throw TypeError('Hop instance is undefined! Please initialize Hop')
-}
-const init = (signer: Signer, chainId: number, toChainId: number) => {
-  const isChainTest = supportedTestnetChains.includes(chainId) ? true : false
-  const isToChainTest = supportedTestnetChains.includes(toChainId)
-    ? true
-    : false
-  // goerli <-> mumbai
-  if (isChainTest && isToChainTest) {
-    hop = new Hop('goerli')
-  } else {
-    hop = new Hop('mainnet')
-  }
-  bridges = {
-    USDT: hop.connect(signer).bridge('USDT'),
-    USDC: hop.connect(signer).bridge('USDC'),
-    MATIC: hop.connect(signer).bridge('MATIC'),
-    DAI: hop.connect(signer).bridge('DAI'),
-    ETH: hop.connect(signer).bridge('ETH'),
+  if (!chain) {
+    throw new Error(`Unsupported chainId passed: ${chainId}`)
   }
+
+  return `https://api.thegraph.com/subgraphs/name/hop-protocol/hop-${chain}`
 }
 
-const getHopBridge = (bridgeCoin: CoinKey) => {
-  isInitialized()
-  if (!Object.keys(bridges).length) {
-    throw Error(
-      'No HopBridge available! Initialize Hop implementation first via init(signer: JsonRpcSigner, chainId: number, toChainId: number)'
-    )
-  }
-  return bridges[bridgeCoin]
+const getTransferIdByTxHash = async (
+  chainId: ChainId,
+  txHash: string
+): Promise<string> => {
+  const url = getSubgraphUrl(chainId)
+  const transfersQuery = `
+  query ($txHash: String) {
+    transfers: transferSents(where: { transactionHash: $txHash } subgraphError: allow) {
+      transferId
+    }
+  }`
+
+  const result = await axios.post<{
+    data: { transfers: { transferId: string }[] }
+  }>(url, {
+    query: transfersQuery,
+    variables: { txHash },
+  })
+
+  return result.data.data.transfers[0]?.transferId
 }
 
-const setAllowanceAndCrossChains = async (
-  bridgeCoin: CoinKey,
-  amount: string,
-  fromChainId: number,
-  toChainId: number
-) => {
-  isInitialized()
-  const bridge = getHopBridge(bridgeCoin)
-  const hopFromChain = hopChains[fromChainId]
-  const hopToChain = hopChains[toChainId]
-  const tx = await bridge.approveAndSend(amount, hopFromChain, hopToChain)
-  return tx
+const getTxHashByTransferId = async (
```

Comment:
The function name should tell that this only works for withdraws.

Reply:
Good point